### PR TITLE
Thread IDs through the compiler pipelines

### PR DESCRIPTION
**Pull request by kustosz**
### Pull Request Description
This plucks IDs defined in the source metadata section and threads them through the compiler pipeline such they are available to the interpreter. This is vital for the language server to properly support execution.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/runtime/src/main/scala/org/enso/compiler/codegen/AstToIR.scala [hunk 1]
```diff
@@ -58,11 +60,13 @@ object AstToIR {
...
           case List(expr) => Some(expr)
           case _ =>
+            val locations    = expressions.map(_.location.map(_.location))
+            val locationsSum = Foldable[List].fold(locations)
             Some(
               Expression.Block(
                 expressions.dropRight(1),
                 expressions.last,
-                location = Foldable[List].foldMap(expressions)(_.location)
+                location = locationsSum.map(IdentifiedLocation(_))
```
Comment (4e6) [inline engine/runtime/src/main/scala/org/enso/compiler/codegen/AstToIR.scala]: Here it's not obvious, that we are dropping location ids from AST. I do realize that collecting ids of expressions kinda doesn't make sense for a block. Ids are not essential for blocks?

File: engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala [hunk 2]
```diff
@@ -1089,7 +1125,8 @@ object IR {
     ) extends Error
         with Kind.Static
         with IRKind.Primitive {
-      override val location: Option[Location] = ast.location
+      override val location: Option[IdentifiedLocation] =
+        ast.location.map(IdentifiedLocation(_))
```
Comment (4e6) [inline engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala]: We don't need AST id for this expression?

File: engine/runtime/src/main/scala/org/enso/compiler/codegen/AstToIR.scala (hunk 1 above)
Comment (4e6) [reply engine/runtime/src/main/scala/org/enso/compiler/codegen/AstToIR.scala]: I just read a comment that blocks should be removed :ok_hand:

Comment (kustosz) [reply engine/runtime/src/main/scala/org/enso/compiler/codegen/AstToIR.scala]: It's not that. It's just that this is the `translateInline` method, called for the inline (i.e. debugger/repl-like flows) where metadata is not really supposed to be present. And it only triggers for the top-level of such expression.

File: engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala (hunk 2 above)
Comment (kustosz) [reply engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala]: Good point. fixed.

**Pull request merged**